app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalSearchController.php

```diff
@@ -50,26 +52,48 @@ class KnowledgeManagementPortalSearchController extends Controller
 {
     public function get(Request $request): KnowledgeManagementSearchData
     {
+        $search = str(json_decode($request->get('search')))
+            ->lower()
+            ->trim();
+
+        $tags = str($request->get('tags'))
+            ->trim()
+            ->when(
+                fn (Stringable $string) => $string->isEmpty(),
+                fn () => collect(),
+                fn (Stringable $string) => $string->explode(',')
+            );
+
         $itemData = KnowledgeBaseArticleData::collection(
             KnowledgeBaseItem::query()
                 ->public()
-                ->tap(new SearchBy('title', Str::lower($request->get('search'))))
+                ->when(Feature::active('tags'), fn (Builder $query) => $query->with('tags'))
+                ->when($search->isNotEmpty(), fn (Builder $query) => $query->tap(new SearchBy('title', $search)))
+                ->when(Feature::active('tags'), fn (Builder $query) => $query->when($tags->isNotEmpty(), fn (Builder $query) => $query->whereHas('tags', fn (Builder $query) => $query->whereIn('id', $tags))))
                 ->get()
-                ->map(function ($item) {
+                ->map(function (KnowledgeBaseItem $article) {
                     return [
-                        'id' => $item->getKey(),
-                        'categoryId' => $item->category_id,
-                        'name' => $item->title,
+                        'id' => $article->getKey(),
+                        'categoryId' => $article->category_id,
+                        'name' => $article->title,
+                        'tags' => Feature::active('tags') ? $article->tags()
+                            ->orderBy('name')
+                            ->get()
+                            ->select([
+                                'id',
+                                'name',
+                            ])
```

Comment:
I think if you put the `select` before the `get` here it will make sure to actually scope the query to the DB, reducing the data in SQL.